v0/destinations/appcues/transform.js

```diff
@@ -0,0 +1,141 @@
+const { EventType } = require("../../../constants");
+
+const {
+  constructPayload,
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  getValueFromMessage,
+  removeUndefinedAndNullValues,
+  removeUndefinedAndNullAndEmptyValues
+} = require("../../util");
+
+const { ConfigCategory, mappingConfig, getEndpoint } = require("./config");
+
+function buildResponse(payload, endpoint) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return {
+    ...response,
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
+}
+
+function processTrackEvent(message, destination, mappingJson) {
+  const requestJson = {};
+  if (message.messageId) {
+    requestJson.request_id = message.messageId;
+  }
+
+  const eventJson = constructPayload(message, mappingJson);
+  requestJson.events = [eventJson];
+
+  return buildResponse(
+    requestJson,
+    getEndpoint(
+      destination.Config.accountId,
+      getFieldValueFromMessage(message, "userIdOnly")
+    )
+  );
+}
+
+function processPage(
+  message,
+  destination,
+  trackMappingJson,
+  profileMappingJson
+) {
+  const requestJson = {};
+  if (message.messageId) {
+    requestJson.request_id = message.messageId;
+  }
+
+  // generating profile part of the payload
+  const profileJson = constructPayload(message, profileMappingJson);
+  // eslint-disable-next-line no-underscore-dangle
+  profileJson._appcuesId = destination.Config.accountId;
+  requestJson.profile_update = profileJson;
+
+  // generating event part of the payload
+  const eventJson = constructPayload(message, trackMappingJson);
+  // eslint-disable-next-line no-underscore-dangle
+  eventJson.attributes._identity = {};
+  // eslint-disable-next-line no-underscore-dangle
+  eventJson.attributes._identity.userId = getFieldValueFromMessage(
+    message,
+    "userIdOnly"
+  );
```

Comment:
```suggestion
  eventJson.attributes = {
    _identity: {
      userId: getFieldValueFromMessage(
        message,
        "userIdOnly"
      );
    }
  };
```

Just a suggestion

Reply:
Its actually a great change, loved it and also used it in other places